src/ui/ControlDescription.tsx

```diff
@@ -0,0 +1,175 @@
+        status: "error",
+        duration: 5000,
+        isClosable: true,
+      });
+      return;
+    }
+    context.describeSelectedEndpoints(context.selectedEndpoints);
+  };
+    
+  return (
+    <>
+      <VStack spacing={4} marginBottom="10em">
+
+        <div className={classes.wrapper}>
+          
+          <Heading as="h2" size="md" marginBottom="1em" marginTop="1em">
+            Select Endpoints for AI Description
+          </Heading>
+          <Text as="h2" size="sm" marginBottom="1em" marginTop="1em">
+            Your OpenAI key is only ever stored locally
+          </Text>
+          <Input
+            placeholder="Enter your OpenAI API Key..."
+            value={apiKey}
+            onChange={(e) => handleApiKeyChange(e.target.value)}
+            className={classes.apiKeyInput}
+            marginBottom="1em"
+          />
+
+          {context.endpoints.length !== 0 && (
+              <Text mt={2} color="gray.700" marginBottom="1em">
+                 Click hostname to select endpoints, then click `Describe Endpoints`
+              </Text>
+            )}
+          {context.endpoints.length === 0 && (
+            <Text mt={2} color="gray.700" marginBottom="1em">
+              No endpoints found. Activate endpoints in browser to start describing them.
+            </Text>
+          )}
+            <Accordion allowMultiple>
+              {Object.entries(endpointsByHost).map(([host, endpoints]) => (
+                <AccordionItem key={host}>
+                  <h2>
+                    <AccordionButton>
+                      <Box flex="1" textAlign="left">{host}</Box>
+                      <AccordionIcon />
+                    </AccordionButton>
+                  </h2>
+                  <AccordionPanel pb={4}>
+                    <div className={classes.wrapper}>
+                      <CheckboxGroup
+                        colorScheme="green"
+                        value={Array.from(context.selectedEndpoints)}
+                        onChange={(values) =>
+                          context.setSelectedEndpoints(new Set(values.map(value => String(value))))
+                        }
+                      >
+                        <VStack align="stretch" spacing={4}>
+                          {endpoints.map((endpoint) => {
+                            const endpointKey = `${endpoint.host}${endpoint.pathname}`;
+                            const tokenCount = context.endpointTokenCounts[endpointKey] || 0;
+                            return (
+                              <Checkbox
+                                key={`${endpoint.host}${endpoint.pathname}`}
+                                value={`${endpoint.host}${endpoint.pathname}`}
+                                onChange={() => onCheckboxChange(`${endpoint.host}${endpoint.pathname}`)}
+                              >
+                                {endpoint.pathname} | Tokens: {tokenCount}
```

Comment:
Personally, I would give the cost in dollars rather than the token count - I think this is a number people are more comfortable with